dir/fs.go

```diff
@@ -92,10 +90,16 @@ func (u unionDirFS) GetPath(elem ...string) (string, error) {
 		}
 		// got the first existing file path and break
 		// return the path with current OS separator
-		return targetPath, nil
+		return filepath.Join(dir.Root(), pathSuffix), nil
 	}
-	// return the last possible path for creating new file
-	return targetPath, &fs.PathError{
+	// if the given path does not exist, return the first possible path in the
+	// union directories for creating new file.
```

Comment:
It still returns an error when falling back. Is that expected?

Reply:
Yes, it is expected. It returns fs.ErrNotExist, so the user know the situation and decide whether report an error or create a new file.